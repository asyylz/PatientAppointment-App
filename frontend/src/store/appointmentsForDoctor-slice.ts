import {
  fetchEntitiesWithIdAndToken,
  createEntitySlice,
} from './create-generic-slice';

export const doctorAppointmentsUrl = (id: string) => {
  return `http://localhost:3000/api/v1/appointments/doctors/${id}`;
};

export const fetchAppointmentsForDoctor =
  fetchEntitiesWithIdAndToken<AppointmentsForDoctor>(
    'appointmentsForDoctor',
    doctorAppointmentsUrl
  );

const appointmentsForDoctorSlice = createEntitySlice<AppointmentsForDoctor>(
  'appointmentsForDoctor',
  fetchAppointmentsForDoctor
);

const { actions, reducer } = appointmentsForDoctorSlice;

export const appointmentsForDoctorActions = actions;
export default reducer;
