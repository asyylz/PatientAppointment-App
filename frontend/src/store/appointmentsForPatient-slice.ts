import {
  fetchEntitiesWithIdAndToken,
  createEntitySlice,
} from './create-generic-slice';


export const patientAppointmentsUrl = (id: string, pagination?: number) => {
  return `http://localhost:3000/api/v1/appointments/patients/${id}?limit=10&page=${pagination}`;
};

export const fetchAppointmentsForPatient =
  fetchEntitiesWithIdAndToken<AppointmentsForPatient>(
    'appointmentsForPatient',
    patientAppointmentsUrl
  );

const appointmentsForPatientSlice = createEntitySlice<AppointmentsForPatient>(
  'appointmentsForPatient',
  fetchAppointmentsForPatient
);

const { actions, reducer } = appointmentsForPatientSlice;

export const appointmentsForPatientActions = actions;
export default reducer;
