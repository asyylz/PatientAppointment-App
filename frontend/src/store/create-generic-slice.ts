import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit';
import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
import { createAxiosInstance } from '../helper/axiosInstance';

/* ------------------------------------------------------ */
/*                        ENTITIES                        */
/* ------------------------------------------------------ */
type HttpMethod = 'get' | 'post' | 'patch' | 'delete';

export const fetchEntities = <T>(
  entity: string,
  url: string,
  method: HttpMethod = 'get'
) =>
  createAsyncThunk<T | object, object | void>(
    `${entity}/fetch`,
    async (arg) => {
      try {
        const axiosMethods: Record<
          HttpMethod,
          (url: string, config?: AxiosRequestConfig) => Promise<AxiosResponse>
        > = {
          get: axios.get,
          post: (url, config) => axios.post(url, arg, config),
          patch: (url, config) => axios.patch(url, arg, config),
          delete: axios.delete,
        };
        const response = await axiosMethods[method](url);
        console.log(`${entity} data:`, response.data.data);
        return response.data.data[entity];
      } catch (err) {
        console.log(err);
      }
    }
  );

/* ------------------------------------------------------ */
/*                    ENTITIES WITH ID                    */
/* ------------------------------------------------------ */
export const fetchEntitiesWithId = <T>(
  entity: string,
  url: (id: string) => string
) =>
  createAsyncThunk<T, string>(`${entity}/fetchWithId`, async (id: string) => {
    const response = await axios.get(url(id));
    return response.data.data[entity]; // Assuming the data is under the entity property
  });

/* ------------------------------------------------------ */
/*            ENTITIES WITH TOKEN AND ID                  */
/* ------------------------------------------------------ */
export const fetchEntitiesWithIdAndToken = <T>(
  entity: string,
  url: (id: string) => string
) =>
  createAsyncThunk<T, { id: string; token: string }>(
    `${entity}/fetchWithIdAndToken`,
    async ({ id, token }) => {
      console.log(entity);
      const response = await axios.get(url(id), {
        headers: { Authorization: `Bearer ${token}` },
      });

      // return response.data.data[entity];
      return response.data.data;
    }
  );

/* ------------------------------------------------------ */
/*                     CREATING SLICE                     */
/* ------------------------------------------------------ */
export const createEntitySlice = <T>(
  entity: string,
  fetchEntityThunk: any,
  additionalReducers?: (builder: any) => void
) => {
  const initialState:
    | ExtendedEntityState<T>
    | EntityState<T>
    | EntityStateForUser<T> = {
    entities: [],
    status: 'idle',
    error: null,
  };

  return createSlice({
    name: entity,
    initialState,
    reducers: {
      addEntity: (state, action: PayloadAction<T>) => {
        state.entities.push(action.payload);
      },
    },
    extraReducers: (builder) => {
      builder
        .addCase(fetchEntityThunk.pending, (state) => {
          state.status = 'loading';
        })
        .addCase(
          fetchEntityThunk.fulfilled,
          (state, action: PayloadAction<T | object>) => {
            state.status = 'succeeded';
            state.entities = action.payload;
            state.error = null;

            console.log('State updated:', state.entities); // Debug state update
          }
        )
        .addCase(fetchEntityThunk.rejected, (state, action) => {
          state.status = 'failed';
          state.error = action.error.message || `Failed to fetch ${entity}`;
        });

      // Apply additional reducers if provided
      if (additionalReducers) {
        additionalReducers(builder);
      }
    },
  });
};

export default createEntitySlice;
