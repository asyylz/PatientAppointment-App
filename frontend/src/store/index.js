import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
  reducer: { departments: departmentsSlice.reducer }, // root reducer
});

export default store;
