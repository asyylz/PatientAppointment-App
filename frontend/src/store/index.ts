import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import departmentsReducer from './departments-slice';
import doctorsReducer from './doctors-slice';
import reviewsReducer from './reviews-slice';
import currentUserReducer from './currentUser-slice';
//import { useDispatch } from 'react-redux';
import searchReducer from './search-slice';
import appointmentsForDoctorReducer from './appointmentsForDoctor-slice';
import appointmentsForPatientReducer from './appointmentsForPatient-slice';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web

// const rootPersistConfig = {
//   key: 'root',
//   storage,
// };

// const persistedReducer = persistReducer(rootPersistConfig, currentUserReducer);

// const store = configureStore({
//   reducer: {
//     currentUser: persistedReducer,
//     departments: departmentsReducer,
//     doctors: doctorsReducer,
//     reviews: reviewsReducer,
//     search: searchReducer,
//     appointmentsForDoctor: appointmentsForDoctorReducer,
//     appointmentsForPatient: appointmentsForPatientReducer,
//   },
//   middleware: (getDefaultMiddleware) =>
//     getDefaultMiddleware({
//       serializableCheck: {
//         ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
//         ignoredActionPaths: ['meta.arg', 'payload.timestamp'],
//         ignoredPaths: ['items.dates'],
//       },
//     }),
//   devTools: process.env.NODE_ENV !== 'production',
// });

// const rootPersistConfig = {
//   key: 'root',
//   storage,
//   whitelist: ['currentUser', 'doctors'], // Only persist the 'currentUserSlice' slice
// };

const authPersistConfig = {
  key: 'currentUser',
  storage: storage,
  // blacklist: ['status', 'error'],
};
const doctorPersistConfig = {
  key: 'doctors',
  storage: storage,
  // blacklist: ['status','error']
};
export const allReducers = combineReducers({
  currentUser: persistReducer(authPersistConfig, currentUserReducer),
  departments: departmentsReducer,
  doctors: persistReducer(doctorPersistConfig, doctorsReducer),
  reviews: reviewsReducer,
  search: searchReducer,
  appointmentsForDoctor: appointmentsForDoctorReducer,
  appointmentsForPatient: appointmentsForPatientReducer,
});

//const persistedReducer = persistReducer(rootPersistConfig, allReducers);

const store = configureStore({
  reducer: allReducers,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;
//export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;

export const persistor = persistStore(store);

export default store;
