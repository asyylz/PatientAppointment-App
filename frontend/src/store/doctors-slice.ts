import { fetchEntities, createEntitySlice } from './create-generic-slice';
import { PayloadAction } from '@reduxjs/toolkit';

// export const doctorsUrl = (pagination?: number, departmentId?: string) => {
//   return `http://localhost:3000/api/v1/doctors?limit=2&page=${pagination}&sort=firstName${
//     departmentId ? `&departmentId=${departmentId}` : ''
//   }`;
// };

export const doctorsUrl = (url: string) => {
  return url ;
};

// Fetch doctors thunk
export const fetchDoctors = fetchEntities<Doctor>('doctors', 'get');

// Create doctors slice
const doctorsSlice = createEntitySlice<Doctor>(
  'doctors',
  fetchDoctors,
  (builder) => {
    builder.addCase(
      'doctors/selectDoctor',
      (state: ExtendedEntityState<Doctor>, action: PayloadAction<Doctor>) => {
        state.selectedDoctor = action.payload;
        console.log('State.selectedDoctor updated:', state.selectedDoctor); // Debug state update
      }
    );
  }
);

// Export the actions and reducer
const { actions, reducer } = doctorsSlice;

export const doctorActions = {
  ...actions,
  selectDoctor: (doctor: Doctor) => ({
    type: 'doctors/selectDoctor',
    payload: doctor,
  }),
};

export default reducer;
