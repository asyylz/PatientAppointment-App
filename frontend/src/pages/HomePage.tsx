import { useEffect, useState } from 'react';
import classes from './HomePage.module.css';
import { useSelector } from 'react-redux';
import { fetchDoctors } from './../store/doctors-slice';
import { useDispatch } from 'react-redux';
import { AppDispatch } from '../store';

const images: { [key: string]: string } = {
  one: 'https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy',
  two: 'https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D',
  three:
    'https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy',
  four: 'https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  five: 'https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D',
};
export default function HomePage() {
  const dispatch: AppDispatch = useDispatch();
  const {
    entities: doctors,
    // status,
    // error,
  } = useSelector((state: RootState) => state.doctors);

  const [currentImageIndex, setCurrentImageIndex] = useState(0);

  useEffect(() => {
    dispatch(
      fetchDoctors(
        'http://localhost:3000/api/v1/doctors?limit=4&page=1&sort=firstName'
      )
    );
  }, [dispatch]);

  useEffect(() => {
    const imageKeys = Object.keys(images);
    const interval = setInterval(() => {
      setCurrentImageIndex((prevIndex) => (prevIndex + 1) % imageKeys.length);
    }, 3000);

    return () => clearInterval(interval);
  }, []);
  // console.log(token);
  return (
    <div className={classes.wrapper}>
      <div className={`${classes.container} ${classes['container--body']}`}>
      
        <section
          className={`${classes.container} ${classes['container--section1']}`}
        >
          <div className={classes.description}>
            <div className={classes['description__item--one']}></div>
            <div className={classes['description__item--two']}>
              <h1 className={classes['description__title']}>
                THE LEADER IN CARING YOU...
              </h1>
              <p className={classes['description__text']}>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos
                maiores possimus rerum delectus magni cupiditate nostrum, ipsa,
                quasi odio facere fuga ea dicta cum debitis in ducimus
                doloremque quod magnam.
              </p>
            </div>
          </div>
        </section>
        <section
          className={`${classes.container} ${classes['container--section2']}`}
        >
          <h1 className={classes['section2__title']}>DISCOVER </h1>
          <a className={classes['section2__title--underline']}>MORE</a>
        </section>
        <section className={classes.galeria}>
          <div className={classes['galeria__images']}>
            {Object.entries(images).map(([key, value], index) => (
              <img
                className={`${classes['galeria__image']} ${
                  classes[`galeria__image--${key}`]
                } ${
                  index === currentImageIndex
                    ? classes['galeria__image--fade']
                    : ''
                }`}
                key={key}
                src={value}
                alt={key}
                loading="lazy"
              />
            ))}
          </div>
        </section>
        <section
          className={`${classes.container} ${classes['container--section3']}`}
        >
          <h1 className={classes['section3__title']}>MEET OUR DOCTORS</h1>
        </section>
        <section
          className={`${classes.container} ${classes['container--section4']}`}
        >
          <div className={classes['section4--carousel']}>
            {doctors.map((doctor: Doctor) => (
              <div className={classes['carousel__card']} key={doctor.id}>
                <div className={classes['carousel__picture']}>
                  <img
                    src={
                      doctor.userId?.image
                        ? doctor.userId?.image
                        : 'https://hips.hearstapps.com/hmg-prod/images/portrait-of-a-happy-young-doctor-in-his-clinic-royalty-free-image-1661432441.jpg?crop=0.66698xw:1xh;center,top&resize=1200:*'
                    }
                    alt={`Dr ${doctor.firstName} ${doctor.lastName}`}
                  />
                </div>
                <div className={classes['carousel__info']}>
                  <h1
                    className={classes['carousel__doctorName']}
                  >{`Dr ${doctor.firstName} ${doctor.lastName}`}</h1>
                  <h1 className={classes['carousel__doctorDepartment']}>
                    {doctor.departmentId.departmentMain}
                  </h1>
                </div>
              </div>
            ))}
          </div>
        </section>
      </div>
    </div>
  );
}
