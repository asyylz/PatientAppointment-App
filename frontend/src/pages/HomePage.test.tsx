import renderer from 'react-test-renderer';
import HomePage from './HomePage'; // Make sure this path is correct
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import store from '../store/index';
import { render, act, screen } from '@testing-library/react';
import { fetchDoctors } from '../store/doctors-slice';

// Mock useEffect globally
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn(),
}));

describe('HomePage snapshot', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  /* -------------------------- - ------------------------- */
  it('1--Renders correctly before data is fetched', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <HomePage />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  /* -------------------------- - ------------------------- */
  it('2--Initializes state with status "idle', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );
    // Initial state should be 'idle'
    expect(store.getState().doctors.status).toEqual('idle');
    expect(store.getState().doctors).toEqual({
      entities: [],
      status: 'idle',
      error: null,
      _persist: { version: -1, rehydrated: true },
    });
  });
  /* -------------------------- - ------------------------- */
  it('3--Dispatches fetchEntities action and updates state with the action payload', async () => {
    expect(store.getState().doctors.status).toEqual('idle');
    render(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );

    await act(async () => {
      store.dispatch(fetchDoctors('mockURL'));
    });

    expect(store.getState().doctors.status).toEqual('loading');
    //console.log(store.getState().doctors.status);

    const fullfilledAction = {
      type: 'doctors/fetch/fulfilled',
      payload: [
        {
          _id: '665f0f2959c971659af920e5',
          gender: 'F',
          image: '/img/doctors/drmathai_small.jpg',
          phone: '647-722-2370',
          address: {
            street: '390 Steeles Avenue West',
            city: 'Vaughan',
            prov: 'ON',
            postal: 'L4J',
            _id: '665f0f2959c971659af920e7',
          },
          firstName: 'Aytekin',
          lastName: 'Yaliz',
          departmentId: {
            _id: '6660fbd12e3c00fe18cfceec',
            departmentMain: 'Allergy & Immunology',
          },
          userId: {
            _id: '666c1f10ad015e6df17a98e4',
            image:
              'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg',
          },
          reviews: [],
        },
      ],
    };

    await act(async () => {
      store.dispatch(fullfilledAction);
    });
    //console.log(store.getState().doctors.status);
    //console.log(store.getState().doctors.entities);
  });
  /* -------------------------- - ------------------------- */
  it('4--Renders fetched data in the DOM after dispatching action', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );

    // After dispatching the action, the state should be updated
    expect(store.getState().doctors).toEqual({
      entities: [
        {
          _id: '665f0f2959c971659af920e5',
          gender: 'F',
          image: '/img/doctors/drmathai_small.jpg',
          phone: '647-722-2370',
          address: {
            street: '390 Steeles Avenue West',
            city: 'Vaughan',
            prov: 'ON',
            postal: 'L4J',
            _id: '665f0f2959c971659af920e7',
          },
          firstName: 'Aytekin',
          lastName: 'Yaliz',
          departmentId: {
            _id: '6660fbd12e3c00fe18cfceec',
            departmentMain: 'Allergy & Immunology',
          },
          userId: {
            _id: '666c1f10ad015e6df17a98e4',
            image:
              'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg',
          },
          reviews: [],
        },
      ],
      status: 'succeeded',
      error: null,
      _persist: { version: -1, rehydrated: true },
    });
  });
  /* -------------------------- - ------------------------- */
  it('5--Displays doctors correctly after data is successfully fetched', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <HomePage />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  /* -------------------------- - ------------------------- */
  it('6--Displays doctor name and department', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <HomePage />
        </MemoryRouter>
      </Provider>
    );
    expect(screen.getByText('Dr Aytekin Yaliz')).toBeInTheDocument();
    expect(screen.getByText('Allergy & Immunology')).toBeInTheDocument();
    expect(screen.getByAltText('Dr Aytekin Yaliz')).toHaveAttribute(
      'src',
      'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg'
    );
    expect(screen.getByAltText('Dr Aytekin Yaliz')).not.toHaveAttribute(
      'src',
      'https://hips.hearstapps.com/hmg-prod/images/portrait-of-a-happy-young-doctor-in-his-clinic-royalty-free-image-1661432441.jpg?crop=0.66698xw:1xh;center,top&resize=1200:*'
    );
  });
});
