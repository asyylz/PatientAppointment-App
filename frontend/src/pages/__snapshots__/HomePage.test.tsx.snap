// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1--Renders correctly before data is fetched 1`] = `
<div
  className="wrapper"
>
  <div
    className="container container--body"
  >
    <section
      className="container container--section1"
    >
      <div
        className="description"
      >
        <div
          className="description__item--one"
        />
        <div
          className="description__item--two"
        >
          <h1
            className="description__title"
          >
            THE LEADER IN CARING YOU...
          </h1>
          <p
            className="description__text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos maiores possimus rerum delectus magni cupiditate nostrum, ipsa, quasi odio facere fuga ea dicta cum debitis in ducimus doloremque quod magnam.
          </p>
        </div>
      </div>
    </section>
    <section
      className="container container--section2"
    >
      <h1
        className="section2__title"
      >
        DISCOVER 
      </h1>
      <a
        className="section2__title--underline"
      >
        MORE
      </a>
    </section>
    <section
      className="galeria"
    >
      <div
        className="galeria__images"
      >
        <img
          alt="one"
          className="galeria__image galeria__image--one galeria__image--fade"
          data-testid="current-image"
          loading="lazy"
          src="https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy"
        />
        <img
          alt="two"
          className="galeria__image galeria__image--two "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D"
        />
        <img
          alt="three"
          className="galeria__image galeria__image--three "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy"
        />
        <img
          alt="four"
          className="galeria__image galeria__image--four "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
        />
        <img
          alt="five"
          className="galeria__image galeria__image--five "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D"
        />
      </div>
    </section>
    <section
      className="container container--section3"
    >
      <h1
        className="section3__title"
      >
        MEET OUR DOCTORS
      </h1>
    </section>
    <section
      className="container container--section4"
    >
      <div
        className="section4--carousel"
      />
    </section>
  </div>
</div>
`;

exports[`HomePage snapshot 5--Displays doctors correctly after data is successfully fetched 1`] = `
<div
  className="wrapper"
>
  <div
    className="container container--body"
  >
    <section
      className="container container--section1"
    >
      <div
        className="description"
      >
        <div
          className="description__item--one"
        />
        <div
          className="description__item--two"
        >
          <h1
            className="description__title"
          >
            THE LEADER IN CARING YOU...
          </h1>
          <p
            className="description__text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos maiores possimus rerum delectus magni cupiditate nostrum, ipsa, quasi odio facere fuga ea dicta cum debitis in ducimus doloremque quod magnam.
          </p>
        </div>
      </div>
    </section>
    <section
      className="container container--section2"
    >
      <h1
        className="section2__title"
      >
        DISCOVER 
      </h1>
      <a
        className="section2__title--underline"
      >
        MORE
      </a>
    </section>
    <section
      className="galeria"
    >
      <div
        className="galeria__images"
      >
        <img
          alt="one"
          className="galeria__image galeria__image--one galeria__image--fade"
          data-testid="current-image"
          loading="lazy"
          src="https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy"
        />
        <img
          alt="two"
          className="galeria__image galeria__image--two "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D"
        />
        <img
          alt="three"
          className="galeria__image galeria__image--three "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy"
        />
        <img
          alt="four"
          className="galeria__image galeria__image--four "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
        />
        <img
          alt="five"
          className="galeria__image galeria__image--five "
          data-testid=""
          loading="lazy"
          src="https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D"
        />
      </div>
    </section>
    <section
      className="container container--section3"
    >
      <h1
        className="section3__title"
      >
        MEET OUR DOCTORS
      </h1>
    </section>
    <section
      className="container container--section4"
    >
      <div
        className="section4--carousel"
      >
        <div
          className="carousel__card"
        >
          <div
            className="carousel__picture"
          >
            <img
              alt="Dr Aytekin Yaliz"
              src="https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg"
            />
          </div>
          <div
            className="carousel__info"
          >
            <h1
              className="carousel__doctorName"
            >
              Dr Aytekin Yaliz
            </h1>
            <h1
              className="carousel__doctorDepartment"
            >
              Allergy & Immunology
            </h1>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
